core/src/lib.rs

```diff
@@ -830,60 +856,42 @@ pub fn import_tokens_proc_internal<T1: Into<TokenStream2>, T2: Into<TokenStream2
         #(#orig_attrs)
         *
         pub #orig_sig {
+            #inner_sig {
+                #(#orig_stmts)
+                *
+            }
+
             use #mm_path::__private::*;
             use #mm_path::__private::quote::ToTokens;
-            let source_path = match syn::parse::<syn::Path>(#tokens_ident) {
-                Ok(path) => path,
-                Err(e) => return e.to_compile_error().into(),
-            };
-            quote::quote! {
-                #mm_override_path::forward_tokens! {
-                    #pound source_path,
-                    #inner_macro_ident,
-                    #mm_override_path
-                }
-            }.into()
-        }
 
-        #[doc(hidden)]
-        #[proc_macro]
-        pub #inner_sig {
-            #(#orig_stmts)
-            *
-        }
-    })
-}
+            syn::custom_keyword!(__private_macro_magic_tokens_forwarded);
 
-/// Internal implementation for the `#[use_proc]` and `#[use_attr]` attribute macros
-pub fn use_internal<T1: Into<TokenStream2>, T2: Into<TokenStream2>>(
-    attr: T1,
-    tokens: T2,
-    mode: ProcMacroType,
-) -> Result<TokenStream2> {
-    parse2::<Nothing>(attr.into())?;
-    let orig_stmt = parse2::<BasicUseStmt>(tokens.into())?;
-    let orig_path = orig_stmt.path.clone();
-    let orig_attrs = orig_stmt.attrs;
-    let vis = orig_stmt.vis;
-    let ident = &orig_stmt
-        .path
-        .segments
-        .last()
-        .expect("path must have at least one segment")
-        .ident;
-    let hidden_ident = match mode {
-        ProcMacroType::Normal => format_ident!("__import_tokens_proc_{}_inner", ident),
-        ProcMacroType::Attribute => format_ident!("__import_tokens_attr_{}_inner", ident),
-        ProcMacroType::Derive => unimplemented!(),
-    };
-    let mut hidden_path: Path = orig_stmt.path.clone();
-    hidden_path.segments.last_mut().unwrap().ident = hidden_ident;
-    Ok(quote! {
-        #(#orig_attrs)
-        *
-        #vis use #orig_path;
-        #[doc(hidden)]
-        #vis use #hidden_path;
```

Comment:
sorry for having miss it. but as you found out in the substrate update of macro magic, this `use_attr` is still needed considering we still hide the macro behind a `__import_tokens_...` (even if is one macro instead of 2 with this PR)

As long as we use a different name for the token export identifier and the macro declaration then we will need `use_attr` and `use_proc`.

If you want I can bring them back because I broke it.

Otherwise we can instead start to use a unique identifier for the export token, and declare the macro with the exact given name.
Then user will have too re-export this item like any normal rust item. It will no longer be hidden.